### PR TITLE
training mode

### PR DESCRIPTION
**Pull request by akash1810**
Add a training mode to the app. When in this mode:
- You can only use the dev youtube channel
- Created composer pages are embargoed indefinitely

# Why?
- Prevent test videos landing on public, heavily subscribed YouTube channels
- Prevent video page launch
- Publishing is scary!

# User flow
- To enter training mode, visit `/training/on`
- To exit training mode, visit `/training/off`
- Links are on `/training` and on `/help`

# UX
Training mode is denoted by a stripy header:
image

Additionally, `/training` has some information:
image

# How it works
Visiting `/training/on` sets session storage in a cookie and its value is injected onto the DOM for client-side use.

Visiting `/training/off` sets the session storage value to `false`.

After `/training/on` and `/training/off` has set session storage, they redirect to `/` for normal app operations... this results in a small "loading..." flash whilst the React app is loaded, but I think that's fine for now (we could always style that loading screen...).

### CODE REVIEW

File: public/video-ui/styles/components/_advanced.scss [hunk 1]
```diff
@@ -11,10 +11,9 @@
         align-items: center;
         padding: 5px 10px;
         width: 100%;
-        background: $color700Grey;
```
Comment (akash1810) [inline public/video-ui/styles/components/_advanced.scss]: use the background of the toolbar

File: public/video-ui/src/services/VideosApi.js [hunk 2]
```diff
@@ -258,5 +258,22 @@ export default {
...
+  },
+
+  preventPublication(composerId, composerUrl) {
+    function doEmbargoIndefinitely(stage) {
+      return pandaReqwest({
+        url: `${composerUrl}/api/content/${composerId}/${stage}/settings/embargoedIndefinitely`,
+        method: 'put',
+        crossOrigin: true,
+        withCredentials: true,
+        data: `"true"`
```
Comment (akash1810) [inline public/video-ui/src/services/VideosApi.js]: 🤢

File: public/video-ui/src/components/Training/Training.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    const trainingModeState = getStore().getState().config.isTrainingMode
+      ? 'on'
+      : 'off';
+
+    return (
+      <div className="container">
+        <h1>Training mode is currently {trainingModeState}</h1>
+        <ul>
+          <li>
+            <a href="/training/on">Turn on</a>
```
Comment (akash1810) [inline public/video-ui/src/components/Training/Training.js]: cannot be `Link`s as they need to go back to the server

File: public/video-ui/src/util/getComposerData.js [hunk 4]
```diff
@@ -77,6 +78,19 @@ export function getComposerData(video) {
...
   ];
+
+  const isTrainingMode = getStore().getState().config.isTrainingMode;
+
+  // block training content being published
+  const embargoedIndefinately = {
+    name: 'embargoedIndefinitely',
+    value: asBooleanString(true),
+    belongsTo: 'settings'
+  };
```
Comment (Reettaphant) [inline public/video-ui/src/util/getComposerData.js]: Do we need to add this here? We are already embargoing the article on page creation, does this do anything other than also embargo it on subsequent publications?

File: public/video-ui/styles/layout/_topbar.scss [hunk 5]
```diff
@@ -3,6 +3,16 @@ $topbarHeight: 50px;
...
   background: $color700Grey;
   border-bottom: 1px solid $greyBorderColor;
+
+  &--training-mode {
+    background: repeating-linear-gradient(
+        -45deg,
+        $color500Grey,
+        $color500Grey 10px,
+        $color800Grey 10px,
+        $color800Grey 20px
```
Comment (Reettaphant) [inline public/video-ui/styles/layout/_topbar.scss]: Would adding opacity here make the title a little bit more visible when in training mode?

Comment (akash1810) [inline public/video-ui/styles/layout/_topbar.scss]: Unfortunately opacity hits the all the elements 😢 . I've forced the background colour back instead:

image

File: public/video-ui/styles/layout/_topbar.scss [hunk 6]
```diff
@@ -12,6 +26,9 @@ $topbarHeight: 50px;
   input {
     background-color: transparent;
     border: none;
+    &::placeholder {
+      color: $color400Grey;
```
Comment (akash1810) [inline public/video-ui/styles/layout/_topbar.scss]: make the placeholder more legible

**Comment by mbarton** [comment]
Might be a daft question but why not get people to use CODE for training?

**Comment by akash1810** [comment]
> why not get people to use CODE for training?

CODE is a pre-prod environment, there's no guarentee it will be operational 100% of the time. Being operational is important particularly as we train overseas offices.

Also, having to remember a different domain is painful.

File: public/video-ui/src/services/VideosApi.js (hunk 2 above)
Comment (Reettaphant) [inline public/video-ui/src/services/VideosApi.js]: Alternatively you could change the creation method in composer to embargo if the embargo is set to true in the create request body (which should be sent automatically). But that would require making another (simple) pr.

Comment (akash1810) [inline public/video-ui/src/services/VideosApi.js]: oooh yeah, that's a **lot** nicer! I'll do this as a secondary PR.

**Pull request merged**